doc/man3/OPENSSL_s390xcap.pod

```diff
@@ -0,0 +1,224 @@
+=pod
+
+=head1 NAME
+
+OPENSSL_s390xcap - the IBM z processor capabilities vector
+
+=head1 SYNOPSIS
+
+ env OPENSSL_s390xcap=... <application>
+
+=head1 DESCRIPTION
+
+libcrypto supports z/Architecture instruction set extensions. These
+extensions are denoted by individual bits in the capabilities vector.
+When libcrypto is initialized, the bits returned by the STFLE instruction
+and by the QUERY functions are stored in the vector.
+
+To change the set of instructions available to an application, you can
+set the OPENSSL_s390xcap environment variable before you start the
+application. After initialization, the capability vector is ANDed bitwise
+with a mask which is derived from the environment variable.
+
+The environment variable is a semicolon-separated list of tokens which is
+processed from left to right (whitespace is ignored):
+
+ OPENSSL_s390xcap="<tok1>;<tok2>;..."
+
+There are three types of tokens:
+
+=over 4
+
+=item <string>
+
+The name of a processor generation. A bit in the environment variable's
+mask is set to one if and only if the specified processor generation
+implements the corresponding instruction set extension. Possible values
+are z900, z990, z9, z10, z196, zEC12, z13 and z14.
+
+=item <string>:<mask>:<mask>
+
+The name of an instruction followed by two 64-bit masks. The part of the
+environment variable's mask corresponding to the specified instruction is
+set to the specified 128-bit mask. Possible values are kimd, klmd, km, kmc,
+kmac, kmctr, kmo, kmf, prno and kma.
+
+=item stfle:<mask>:<mask>:<mask>
+
+Store-facility-list-extended (stfle) followed by three 64-bit masks. The
+part of the environment variable's mask corresponding to the stfle
+instruction is set to the specified 192-bit mask.
+
+=back
+
+The 64-bit masks are specified in hexadecimal notation. The 0x prefix is
+optional. Prefix a mask with a tilde (~) to denote a bitwise NOT operation.
+
+The following bits are significant ...
+
+=over 4
+
+=item stfle:
+
+ #38 store-clock-fast facility
+ #46 message-security assist
+ :
+ #50 message-security assist extension 4
+ #51 message-security assist extension 3
+ :
+ #45 message-security assist extension 8
```

Comment:
This is confusing in sense that there is no mark when you switch to next word. For example 46 and 50 are from different words. In ia32cap.pod second word is denoted as 64+n. Maybe it would be appropriate even here? The remark naturally applies even to function capability vectors...

Reply:
When listing the significant bits i use the format described above: that is, the 64-bit words/masks are separated by the colons.
Im not sure if adding 64+n notation will make it less or more confusing.. ?

Reply:
Problem of course is that we both know what's going on and only *imagine* how readable would this be to an *unprepared* reader. At the same time reader won't be totally unprepared, as [s]he has to make connections between this text and z/Architecture Principles of Operations. And that's where it gets extra tricky. In the manual bits from multiple words are numbered monotonically, but they are numbered in reverse order. For example vector facility is bit 129. While here it's denoted as 62 (btw, colon is missing, right?). Sigh... Maybe it would be appropriate to make the table three-column, something like
```
#129  #::62 vector facility
```
First column would be the number from manual and second would be right shift that would help to calculate actual values. Values from first word could be denoted with `n::`, from second as `:n:`

Reply:
> `:n:`

Or maybe even ':1<<n:`

Reply:
> (btw, colon is missing, right?)

No, it's not, sorry about distraction.

Reply:
> [s]he has to make connections between this text and z/Architecture Principles of Operations

This is what i tried to avoid by not providing a POP reference in the docs and assuming sane bit ordering.. such that user does not have to do "n=(63 - (n_POP) % 64)" before calculating the corresponding mask 2^n.

> colon is missing, right?

No? vector facility is in third word, together with msa8 and other vector ext.. I mean you have to read the list top to bottom : 

stfle:
<list of 1st word bits>
: 
<list of 2nd word bits>
: 
< list of 3rd word bits>

Your suggestion of a two column table makes sense if you see a need to reference POP bit counting at all.

Reply:
> a need to reference POP bit counting

I'd say it's inevitable. Inquisitive reader would surely like to know what does e.g. "message-security assist" entail, and would have to turn to the manual. And it is now would just get disoriented...

Reply:
One can also look at it as following. Things should be verifiable. And easier it is to verify, better. It's only appropriate to present the POP numbering and bit numbering in word next to each other, so that one can double-check that everything in shape when something doesn't appear right. I mean imagine I assign a variable and get unexpected result. I might need to double-check whole chain from manual to actual assignment...

Reply:
Okay, in this case i ll also add a"see also" section with reference to latest POP.

Id still prefer a list (which reflects the env var format) as described above instead of a table and repeat e.g. "::" in front of all third word bit. what do you think ?

Reply:
mask bits could still be denoted by 1<<n to make it even more clear, what is meant by the non-POP number.

Reply:
Problem I have with `:` is that they are experienced as vertical ellipses, as if there is gap between significant bits, rather than word delimiters. But all right, the fact that there will "64*n-ish" gap in first column it might be ok. E.g.
```
#17     #1<<46   message-security assist
:
#76     #1<<51   message-security assist 3
:
#129   #1<<62   vector facility
```
Might work...